main.js

```diff
@@ -1,16 +1,57 @@
 define(function (require, exports, module) {
 	var fileInfo = {};
 
+	// Load preferences
+	var PreferencesManager = brackets.getModule("preferences/PreferencesManager"),
+		prefs = PreferencesManager.getExtensionPrefs("file-icons");
+	prefs.definePreference("icons", "array", []);
+
+	// Declare variables
+	var iconArray;
+	var iconObject = {};
+	var iconArray = prefs.get("icons");
+
+	//Function to check the Preferences file for the extension
+	//And update the icon if found
+	function checkPrefs(search, a, b, c) {
+		iconObject.icon = a;
+		iconObject.color = b;
+		iconObject.size = c;
+
+		if (iconArray) {
+			for (var i = 0; i < iconArray.length; i++) {
+				var object = iconArray[i];
+				if (search == object["extension"]) {
+					for (var key in object) {
+						if (object[key]) {
+							iconObject[key] = object[key];
+						}
+					}
+				}
+			}
+			return iconObject;
+		}
+	}
+
 	function addIcon(extension, icon, color, size) {
+		checkPrefs(extension, icon, color, size);
```

Comment:
Don't call checkPrefs here, I think there should be one array with the default icons and one with the user icons. This way we can reload the user icons array if the user changes his settings, without needing to restart Brackets.